src/util.lua

```diff
@@ -100,42 +104,89 @@ function util.copyargs(args)
 end
 
 -----------------------------------------------
--- Finds matching arguments in a saved list of arguments
--- @param argslist list of arguments from which to search
--- @param args the arguments of which to find a match
--- @return the matching arguments if a match is found, otherwise nil
-function util.matchargs(argslist, args)
-  local function matches(t1, t2, t1refs)
-    local match = require 'luassert.match'
-    for k1,v1 in pairs(t1) do
-      local v2 = t2[k1]
-      if match.is_matcher(v1) then
-        if not v1(v2) then return false end
-      elseif match.is_matcher(v2) then
-        if match.is_ref_matcher(v2) then v1 = t1refs[k1] end
-        if not v2(v1) then return false end
-      elseif (v2 == nil or not util.deepcompare(v1,v2)) then
+-- Clear an arguments or return values list from a table
+-- @param arglist the table to clear of arguments or return values and their count
+-- @return No return values
+function util.cleararglist(arglist)
+  for idx = #arglist, 1, -1 do
```

Comment:
`#arglist` will fail if there are `nil`s in the table. Should be a numeric loop from `arglist.n`

Reply:
Updated.